camelcase.js

```diff
@@ -0,0 +1,9 @@
+"use strict";
+var sentenceCase = require("./sentenceCase");
+module.exports = function(string){
+    var output = string.match(/\b[^\s]+\b/gi).map(function(value){
```

Comment:
You may be aware already but `String.prototype.replace` can take a [function as the replacement value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/replace#Specifying_a_function_as_a_parameter). It calls the function for each match, and use the function's return value as the new value, very similar to what you are doing here.